src/main/java/org/gitlab4j/api/GitLabApiClient.java

```diff
@@ -600,23 +600,37 @@ protected Response upload(String name, File fileToUpload, String mediaTypeString
      * @throws IOException if an error occurs while constructing the URL
      */
     protected Response upload(String name, File fileToUpload, String mediaTypeString, Form formData, URL url) throws IOException {
+        MediaType mediaType = (mediaTypeString != null ? MediaType.valueOf(mediaTypeString) : null);
+        FileDataBodyPart filePart = mediaType != null ?
+            new FileDataBodyPart(name, fileToUpload, mediaType) :
+            new FileDataBodyPart(name, fileToUpload);
+        return upload(filePart, formData, url);
+    }
 
+    protected Response upload(String name, InputStream inputStream, String filename, String mediaTypeString, Object... pathArgs) throws IOException {
+        URL url = getApiUrl(pathArgs);
+        return (upload(name, inputStream, filename, mediaTypeString, null, url));
+    }
+
+    protected Response upload(String name, InputStream inputStream, String filename, String mediaTypeString, Form formData, URL url) throws IOException {
         MediaType mediaType = (mediaTypeString != null ? MediaType.valueOf(mediaTypeString) : null);
-        try (FormDataMultiPart multiPart = new FormDataMultiPart()) {
+        StreamDataBodyPart streamDataBodyPart = mediaType != null ?
```

Comment:
After checking in `StreamDataBodyPart` constructors, you don't need to test if mediaType is null or not. In fact the constructor used line 619 call the one used line 618 with null in MediaType. Please remove the ternary operator.

Reply:
In this case it's not really the same thing.
As `StreamDataBodyPart`'s javadoc states:

> It sets the InputStream as a body part with the default MediaType.APPLICATION_OCTET_STREAM_TYPE (if not specified by the user).
> Note: The MIME type of the entity cannot be automatically predicted as in case of FileDataBodyPart.

[Gitlab docs](https://docs.gitlab.com/ee/api/projects.html#upload-a-file) do not help clarifying.
But I guess it's not a problem, I think Gitlab REST API doesn't care about the media type.
What do you think @jabby?